fleet/fleet/overrides/timesheet.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2025, AgriTheory and contributors
+# For license information, please see license.txt
+
+from datetime import datetime
+
+import frappe
+
+
+@frappe.whitelist()
+def fetch_timesheet_from_vehicle_log(employee, start_date, end_date):
+	# Fetch all vehicle logs for the employee with geofence entered, ordered by time
+	vehicle_logs = frappe.db.sql(
+		"""
+		SELECT name, employee, creation, geofences_entered, geofences_exited
+		FROM `tabVehicle Log`
+		WHERE employee = %s
+		  AND (
+				(geofences_entered IS NOT NULL AND geofences_entered != '')
+			 OR (geofences_exited IS NOT NULL AND geofences_exited != '')
+		  )
+		  AND creation BETWEEN %s AND %s
+		ORDER BY creation ASC
+		""",
+		(employee, start_date, end_date),
+		as_dict=1,
+	)
+
+	fmt = "%Y-%m-%d %H:%M:%S"
+	results = []
+	i = 0
+	n = len(vehicle_logs)
+	while i < n:
```

Comment:
Can you use a `for row in vehicle_logs` loop here instead? The length of data is known so using `while` and an iterator only makes it hard to read.